test/integration/controller_test.go

```diff
@@ -423,9 +474,18 @@ func TestCAPIClusterCreate(t *testing.T) {
 			}
 
 			machineSets := make([]*capiv1alpha1.MachineSet, len(tc.machineSets))
-			var masterMachinSet *capiv1alpha1.MachineSet
+			var masterMachineSet *capiv1alpha1.MachineSet
+
+			for i, clustopClusterMS := range tc.machineSets {
+
+				role := capicommon.NodeRole
+				if clustopClusterMS.NodeType == clustopv1alpha1.NodeTypeMaster {
+					role = capicommon.MasterRole
+				}
+
+				machineSet := testCAPIMachineSet(clustopClusterMS.ShortName, int32(clustopClusterMS.Size),
```

Comment:
Use a name for the machineset that includes the cluster.
For example, `fmt.Sprintf("%s-%s", cluster.Name, clustopClusterMS.ShortName)`